src/emqx.erl

```diff
@@ -128,36 +128,43 @@ subscribed(Topic, SubPid) when is_pid(SubPid) ->
 subscribed(Topic, SubId) when is_atom(SubId); is_binary(SubId) ->
     emqx_broker:subscribed(iolist_to_binary(Topic), SubId).
 
-%%--------------------------------------------------------------------
+%%------------------------------------------------------------------------------
 %% Hooks API
-%%--------------------------------------------------------------------
-
--spec(hook(atom(), function() | {emqx_hooks:hooktag(), function()}, list(any()))
-      -> ok | {error, term()}).
-hook(Hook, TagFunction, InitArgs) ->
-    emqx_hooks:add(Hook, TagFunction, InitArgs).
-
--spec(hook(atom(), function() | {emqx_hooks:hooktag(), function()}, list(any()), integer())
-      -> ok | {error, term()}).
-hook(Hook, TagFunction, InitArgs, Priority) ->
-    emqx_hooks:add(Hook, TagFunction, InitArgs, Priority).
-
--spec(unhook(atom(), function() | {emqx_hooks:hooktag(), function()})
-      -> ok | {error, term()}).
-unhook(Hook, TagFunction) ->
-    emqx_hooks:delete(Hook, TagFunction).
-
--spec(run_hooks(atom(), list(any())) -> ok | stop).
-run_hooks(Hook, Args) ->
-    emqx_hooks:run(Hook, Args).
-
--spec(run_hooks(atom(), list(any()), any()) -> {ok | stop, any()}).
-run_hooks(Hook, Args, Acc) ->
-    emqx_hooks:run(Hook, Args, Acc).
-
-%%--------------------------------------------------------------------
+%%------------------------------------------------------------------------------
+
+-spec(hook(emqx_hooks:hookpoint(), emqx_hooks:action()) -> ok | {error, already_exists}).
+hook(HookPoint, Action) ->
+    emqx_hooks:add(HookPoint, Action).
+
+-spec(hook(emqx_hooks:hookpoint(), emqx_hooks:action(), emqx_hooks:filter() | integer())
+      -> ok | {error, already_exists}).
```

Comment:
i've seen `=` and `->` being broke out to a new line. is it some kind of convention ?